Core/Prognetics.Saga.Parsers.DependencyInjection/JsonParserExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Prognetics.Saga.Core.Abstract;
+using Prognetics.Saga.Parsers.Core.Model;
+using System.Reflection;
+
+namespace Prognetics.Saga.Parsers.DependencyInjection
+{
+    public static class JsonParserExtensions
+    {
+        public static void RegisterJsonParsers(this IServiceCollection services,
+            Action<ModelSourceOptions> configureOptions)
+        {
+            services.Configure(configureOptions);
+            services.AddTransient(x => x.GetRequiredService<IOptions<ModelSourceOptions>>().Value);
+
+            services.AddSingleton<IEnumerable<IModelSource>>(provider =>
+            {
+                var result = new List<IModelSource>();
+                var options = provider.GetRequiredService<ModelSourceOptions>();
+
+                foreach (var confirguration in options.Configurations)
+                {
+                    var type = Type.GetType(confirguration.ParserType, true);
```

Comment:
Refering to the [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.type.gettype?view=net-7.0#system-type-gettype(system-string)) if type is not in the current executing assembly or in mscorlib AssemblyQualifiedName should be provided. It could be not user friendly to pass every time whole phrase instead of for example just "json-file-parser".

Reply:
Resolved on stream.